st2common/st2common/models/api/trigger.py

```diff
@@ -152,6 +152,10 @@ def to_model(cls, trigger):
                           parameters=parameters)
         return model
 
+    def to_dict(self):
+        # Return dictionary version of the trigger
+        return vars(self)
```

Comment:
I am concerned this might return internal fields but other than explicit construction of the object, I don't see a way out. I think we need this to_dict() for all models too. So a base class abstract method might be better.